File: frontend/src/lib/types/content-tag.ts
// TypeScript types for the Content Tagging system

export interface ContentTag {
  id: number;
  name: string;
  slug: string;
  description?: string;
  category?: string;
  icon?: string;
  color?: string;
  order_index: number;
  is_active: boolean;
  created_at?: string;
  updated_at?: string;
}

export interface ContentTagCreate {
  name: string;
  slug: string;
  description?: string;
  category?: string;
  icon?: string;
  color?: string;
  order_index?: number;
  is_active?: boolean;
}

export interface ContentTagUpdate {
  name?: string;
  slug?: string;
  description?: string;
  category?: string;
  icon?: string;
  color?: string;
  order_index?: number;
  is_active?: boolean;
}

export type TagCategory =
  | 'style'
  | 'vibe'
  | 'activity_type'
  | 'destination_type'
  | 'budget'
  | 'duration'
  | 'general';

File: backend/app/schemas/content_tag.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional, List
from datetime import datetime


class ContentTagBase(BaseModel):
    name: str = Field(..., description="Display name of the tag", example="Family Friendly")
    slug: str = Field(..., description="URL-safe slug", example="family-friendly")
    description: Optional[str] = Field(None, description="Optional description of what this tag means")
    category: Optional[str] = Field(
        None,
        description="Grouping category for the tag. e.g. 'audience', 'pace', 'environment', 'budget'",
        example="audience",
    )
    icon: Optional[str] = Field(None, description="Emoji or icon identifier", example="👨‍👩‍👧")
    color: Optional[str] = Field(None, description="Hex color code for frontend chip", example="#4CAF50")
    order_index: Optional[int] = Field(0, description="Display order (lower = earlier)")


class ContentTagCreate(ContentTagBase):
    is_active: Optional[bool] = Field(True, description="Whether the tag is active")


class ContentTagUpdate(BaseModel):
    name: Optional[str] = Field(None, description="Display name of the tag")
    slug: Optional[str] = Field(None, description="URL-safe slug")
    description: Optional[str] = Field(None, description="Optional description")
    category: Optional[str] = Field(None, description="Grouping category")
    icon: Optional[str] = Field(None, description="Emoji or icon identifier")
    color: Optional[str] = Field(None, description="Hex color code for frontend chip")
    order_index: Optional[int] = Field(None, description="Display order")
    is_active: Optional[bool] = Field(None, description="Whether the tag is active")


class ContentTagResponse(ContentTagBase):
    id: int
    is_active: Optional[bool] = Field(True, description="Whether the tag is active")
    created_at: Optional[datetime] = Field(None, description="Creation timestamp")
    updated_at: Optional[datetime] = Field(None, description="Update timestamp")

    model_config = ConfigDict(from_attributes=True)


class PaginatedContentTagResponse(BaseModel):
    items: List[ContentTagResponse]
    total: int
    page: int
    size: int
    pages: int
